homeassistant/components/tractive/manifest.json

```diff
@@ -0,0 +1,15 @@
+{
+  "domain": "tractive",
+  "name": "Tractive",
+  "config_flow": true,
+  "documentation": "https://www.home-assistant.io/integrations/tractive",
+  "requirements": [
+    "aiotractive==0.5.1"
+  ],
+  "codeowners": [
+    "@Danielhiversen",
+    "@zhulik"
+  ],
+  "iot_class": "cloud_push",
+  "quality_scale": "silver"
```

Comment:
For it to be silver-rated, it needs handle expiration of auth credentials.

> Handles expiration of auth credentials. Refresh if possible or print correct error and fail setup. If based on a config entry, should trigger a new config entry flow to re-authorize.

https://developers.home-assistant.io/docs/integration_quality_scale_index/#silver-%F0%9F%A5%88

Currently, the integration only checks this in the integration config flow, but doesn't have a flow for handling reauth for when e.g., credentials expire later on.

Reply:
It is handled by [aiotractive](https://github.com/zhulik/aiotractive/blob/main/aiotractive/api.py#L85)

Reply:
@zhulik A package cannot offer a home assistant configuration flow to ask the user to update/change their credentials.

Reply:
@frenck got it, could you please point me at an integration that does it properly so I'll use it as an example?

Reply:
Sure, search for config_flow.py's that have `async_step_reauth` implemented

Reply:
I've just gone through a couple of them, just to confirm: I need to wrap all 3rd party authentication errors into `ConfigEntryAuthFailed` and implement `async_step_reauth` in the `config_flow.py`? All the rest will work automagically or I need to take care of something else?

Reply:
I think that is correct.
>  Raising ConfigEntryAuthFailed will cancel future updates

https://developers.home-assistant.io/docs/integration_fetching_data/

Reply:
@zhulik You could also remove the quality scale and add this functionality later.
Would be nice to have this merged.